### PR TITLE
[Redesigned URLS 6] Get all tests passing on windows

### PR DESCRIPTION
**Pull request by rictic**
- [x] CHANGELOG.md not updated, purely bug fixes here

### CODE REVIEW

**Comment by rictic** [comment]
Hm, changes have introduced a minor broken test. Will investigate after dinner.

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Test against the latest version of this Node.js version
+environment:
+  nodejs_version: "8"
```
Comment (justinfagnani) [inline appveyor.yml]: node 9?

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+install:
+  - ps: Install-Product node $env:nodejs_version
+  - npm install
+
+test_script:
+  # Output useful info for debugging.
+  - node --version
+  - npm --version
+  # run tests
```
Comment (justinfagnani) [inline appveyor.yml]: that comment isn't necessary :)

File: gulpfile.js [hunk 3]
```diff
@@ -28,67 +28,71 @@ const mergeStream = require('merge-stream');
...
-        `A task with the name ${JSON.stringify(name)} already exists!`);
+      `A task with the name ${JSON.stringify(name)} already exists!`);
   }
   gulp.task(name, deps, impl);
 }
 
 task('init');
 
-task('depcheck', function() {
+task('depcheck', function () {
```
Comment (justinfagnani) [inline gulpfile.js]: it'd be nice to upgrade to async functions...

File: src/test/url-loader/package-url-resolver_test.ts [hunk 4]
```diff
@@ -13,117 +13,145 @@
...
+/**
+ * On posix systems file urls look like:
+ *      file:///path/to/foo
+ * On windows they look like:
+ *      file:///c%3A/path/to/foo
+ *
+ * This will produce an OS-correct file url.
+ */
+function rootedFileUrl([text]: TemplateStringsArray): ResolvedUrl {
+  const root = URI.file(pathlib.resolve('/')).toString();
```
Comment (justinfagnani) [inline src/test/url-loader/package-url-resolver_test.ts]: Use the `noOpTag` you added before

File: src/url-loader/package-url-resolver.ts [hunk 5]
```diff
@@ -41,16 +43,16 @@ export class PackageUrlResolver extends UrlResolver {
   constructor(options?: PackageUrlResolverOptions) {
     super();
     options = options || {};
-    let packageDir = options.packageDir || process.cwd();
-    if (process.platform === 'win32') {
-      packageDir = packageDir.replace(/\\/g, '/');
+    this.packageDir = pathlib.resolve(options.packageDir || process.cwd());
+    if (isWindows && /^[a-z]:/.test(this.packageDir)) {
+      // Upper case the drive letter
+      this.packageDir = this.packageDir[0].toUpperCase() + this.packageDir.slice(1);
```
Comment (justinfagnani) [inline src/url-loader/package-url-resolver.ts]: looks like we're building up a normalization function here

File: src/url-loader/package-url-resolver.ts [hunk 6]
```diff
@@ -156,13 +159,24 @@ export class PackageUrlResolver extends UrlResolver {
...
+      if (path && path.startsWith(this.resolvedComponentDir)) {
+        return path;
       }
     }
     return undefined;
   }
+
+  private filesystemPathForPathname(decodedPathname: string) {
+    let path = Uri.file(decodedPathname).fsPath;
+    if (isWindows && /^[a-z]:/.test(path)) {
```
Comment (justinfagnani) [inline src/url-loader/package-url-resolver.ts]: duplicate logic from L47

File: src/url-loader/package-url-resolver.ts [hunk 7]
```diff
@@ -156,13 +159,24 @@ export class PackageUrlResolver extends UrlResolver {
...
+      // Upper case the drive letter
+      path = path[0].toUpperCase() + path.slice(1);
```
Comment (justinfagnani) [inline src/url-loader/package-url-resolver.ts]: I find String.replace pretty easy to read for cases like this:

```ts
path.replace(/^([a-z]):/, (_, l) => l.toUpperCase() + ':')
```

**Pull request merged**